### PR TITLE
Clean up URLs in menu, tests, and controller

### PR DESCRIPTION
**Pull request by Scotchester**
Generally related to recent updates to things that have moved from the fin-ed-resources Sheerlike project into Wagtail. Also does some long-overdue updating of the smoke test list.

## Additions

- Adds some recently created pages to the full URL list in `http_smoke_test`

## Removals

- Removes `urls.py` rules for the former fin-ed-resources pages that have (thus far) migrated to Wagtail.

## Changes

- Updated the following mega menu links to point to their canonical URLs:
  - Planning for Retirement
  - Resources for Tax Preparers (just migrated to Wagtail)
  - Free Brochures (now with `https`)
  - Mortgage Database (HMDA)
- Ensured that full URL list in `http_smoke_test` reflected the current contents of the mega menu, in order.

## Testing

- Pull fork, run server, see menu updates.
- I don't know how to actually run the smoke tests locally.

## Todos

- Someone should document the logic behind what goes on the short-run list in `http_smoke_test`, and update the list accordingly.

## Checklist

* [x] PR has an informative and human-readable title
* [x] Changes are limited to a single goal (no scope creep)
* [x] Code can be automatically merged (no conflicts)
* [x] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [x] New functions include new tests
* [x] New functions are documented (with a description, list of inputs, and expected output)
* [x] Placeholder code is flagged
* [x] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated
* [x] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

File: cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html [hunk 1]
```diff
@@ -102,19 +102,19 @@
...
         ('Resources for Libraries', '/library-resources/', []),
-        ('Resources for Tax Preparers', '/tax-preparer-resources/', []),
+        ('Resources for Tax Preparers', '/educational-resources/resources-for-tax-preparers/', []),
         ('Resources for Parents', '/educational-resources/money-as-you-grow/', [])
     ),(
         ('Information for Economically Vulnerable Consumers', '/empowerment/', []),
         ('Resources for Older Adults and Their Families', '/educational-resources/resources-for-older-adults/', []),
-        ('Free Brochures', 'http://pueblo.gpo.gov/CFPBPubs/CFPBPubs.php', [])
+        ('Free Brochures', 'https://pueblo.gpo.gov/CFPBPubs/CFPBPubs.php', [])
     )]),
```
Comment (richaagarwal) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: I know it's out of scope for this PR, but do you have any thoughts on getting this menu content into Wagtail?

Comment (Scotchester) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: I don't have any thoughts on that yet, but definitely interested in exploring the idea!

Comment (chosak) [inline cfgov/jinja2/v1/_includes/organisms/_vars-mega-menu.html]: I've been meaning to look into [wagtailmenus](https://github.com/rkhleics/wagtailmenus) which seems designed for this use case.

**Comment by Scotchester** [comment]
@chosak Seems like that only runs the short-run test. Is there a way I can set `FULL` to `True` when invoking the script?

**Comment by chosak** [comment]
@Scotchester I think adding `--full`, e.g.

```sh
$ python cfgov/scripts/http_smoke_test.py --base http://localhost:8000 -v --full
```

will hit the full set.

**Comment by Scotchester** [comment]
That worked, thanks! :+1:

Everything failing is stuff that I don't have installed locally, so I think this is good to go. Merging once Travis finishes.

**Pull request merged**